instrument/utils/utilities.py

```diff
@@ -0,0 +1,446 @@
+            ) = _geom_.calc._engine.engine.parameters_values_get(
+                Hkl.UnitEnum.USER
+            )
+        else:
+            raise ValueError("Mode '{}' not supported.".format(mode))
+
+        if len(args) == 0:
+            value = _geom_.get_axis_constraints(axis).value
+            value = int(
+                (input("Freeze {} to ({})? ".format(axis, value))) or value
+            )
+        elif len(args) == 1:
+            value = args[0]
+        else:
+            raise ValueError(
+                "either no argument or angle value needs to be provided."
+            )
+        if axis == "psi":
+            _geom_.calc._engine.engine.parameters_values_set(
+                [_h2, _k2, _l2, value], Hkl.UnitEnum.USER
+            )
+        else:
+            ll = _geom_.get_axis_constraints(axis).low_limit
+            hl = _geom_.get_axis_constraints(axis).high_limit
+            _geom_.apply_constraints({axis: Constraint(ll, hl, value, True)})
+        print("{} frozen to {}".format(axis, value))
+    else:
+        raise ValueError("Function not available for mode '{}'".format(mode))
+
+
+def show_constraints():
+    """
+    Show constraints and freeze angles (value)
+    """
+    _geom_ = current_diffractometer()
+    _geom_.show_constraints()
+
+
+def reset_constraints():
+    """
+    Reset all constraints
+    """
+    _geom_ = current_diffractometer()
+    _geom_.reset_constraints()
+    _geom_.show_constraints()
+
+
+def set_constraints():
+    """
+    Change constraint values for specific axis
+    """
+    _geom_ = current_diffractometer()
+    axes = _geom_.calc._engine.engine.axis_names_get(0)
+    for axis in axes:
+        low = _geom_.get_axis_constraints(axis).low_limit
+        high = _geom_.get_axis_constraints(axis).high_limit
+        angle = _geom_.get_axis_constraints(axis).value
+        value = (
+            input(
+                "{} constraints low, high = [{:3.3f}, {:3.3f}]: ".format(
+                    axis, low, high
+                )
+            )
+        ) or [low, high]
+        if isinstance(value, str):
+            value = value.replace(",", " ").split(" ")
+        _geom_.apply_constraints(
+            {axis: Constraint(value[0], value[1], angle, True)}
+        )
+    _geom_.show_constraints()
+
+
+def change_diffractometer(*args):
+    _geom_ = current_diffractometer()
+    list = ["diffract", "fourc", "psic"]
+    print("Available diffractometers {}".format(list))
+    diff = input("Diffractometer ({}): ".format(_geom_.name)) or _geom_.name
+    if diff == "diffract":
+        select_diffractometer(diffract)
+    elif diff == "fourc":
+        select_diffractometer(fourc)
+    elif diff == "psic":
+        select_diffractometer(psic)
+    else:
+        raise ValueError("Diffractometer type {} not existing.".format(diff))
+
+    _geom_ = current_diffractometer()
+    print("Diffractometer changed to {}".format(_geom_.name))
+
+
+def plotselect(detector=None):
```

Comment:
Isn't this mostly already in [counters](https://github.com/APS-4ID-POLAR/polar_instrument/blob/add_simscalar/instrument/utils/counters_class.py)? Maybe just need to add a function there to print the scaler options?